pdpserver/server/query.go

```diff
@@ -0,0 +1,131 @@
+package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/infobloxopen/themis/pdp"
+	"github.com/julienschmidt/httprouter"
+)
+
+const (
+	storageQueryCommands = `PDP Query API
+
+GET /root
+GET /storage/<path...>?depth=<depth> [depth is any value of range 0 .. n. defaults to 0]
+GET /find/<id>/<start path...> [ensure to end with "/" if start path is root]
+`
+	maximumRetries  = 100
+	maxUint32       = 2147483647
+	defaultDepthMsg = "proceeding with depth=0"
+)
+
+func queryIndexHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	fmt.Fprint(w, storageQueryCommands)
+}
+
+func (s *Server) listenQuery() error {
```

Comment:
The function name is misleading and too generic. Too generic because doesn't specify what query it listen for. Should be something like `listenPolicyStorage`. Misleading because it's ListenAndServe. Which it shouldn't be. We currently have a couple of examples how to do it (health check and profile server). Basically, maybe it's good point to gather them all together at single http port.